### DISCUSSION TITLE
livewire-tmp not cleaning up

### DISCUSSION THREAD

**Discussion by Loots-it** [livewire/livewire#3470] [7. Bugs (archived)] [answered]
This folder is not cleaned up, not even after 24 hours (or more). I'm not using S3, just the local filesystem. I'm not sure what the problem is.

Because of this, my backups are getting too large. Of course there is no point in backing up the tmp files so I will have to exclude this folder anyway. But the tmp folder can get quite large in comparison to the other folders because the incoming photos aren't minified yet.

livewire: v2.5.5
laravel 8
php 8
using laravel forge

**Comment by Qirel** [comment, marked as answer]
The `WithFileUploads` trait has a method called [cleanupOldUploads()](https://github.com/livewire/livewire/blob/b0f4026509604f5239813a605ba7172f4bb5fa0e/src/WithFileUploads.php#L99) that runs in the `finishUpload()` method within the same trait. In turn, that method is called from the [UploadManager in JavaScript](https://github.com/livewire/livewire/blob/69575f50bb7f8a49a41f9bd6bd16c73a6ef4fda3/js/component/UploadManager.js#L117) once the file was uploaded.

This means that, in order for the uploads-folder to be cleaned up, you have to upload a new file - the `cleanupOldUploads()` method will then run and clean up files that are older than 24 hours. Apart from that, I don't think there is any magic that deletes files from that folder. 

So the question then becomes - does it not remove uploads that are older than 24 hours even after you upload new files?

**Reply by Loots-it** [reply]
Thanks for your answer! Indeed,  I figured it out myself too.

https://github.com/livewire/livewire/issues/3471

As long as I don't upload any new files, the old ones are not removed. But in my opinion, it would make sense if there was an option to delete the old files in a scheduled way. Not as part of another request of possibly another user.

**Reply by Qirel** [reply]
Should be fairly easy to implement on your own, you can run a scheduled job with a single command.

```php
// Clean the Livewire temp-upload folder
\Illuminate\Support\Facades\File::cleanDirectory(\storage_path('app/livewire-tmp'));
```

**Comment by fantomas-code** [comment]
### \Illuminate\Support\Facades\File::cleanDirectory(\storage_path('app/livewire-tmp'));
Where does this instruction go, or where is it placed?  👆

**Reply by fd6130** [reply]
probably make a cronjob and then run that code?
